### PR TITLE
Gutenberg: Add Jetpack colophon to Jetpack blocks

### LINKED ISSUE #27773: Add Jetpack branding to the Inspector
We need to add a Jetpack colophon of some sort to the block inspector for Jetpack blocks. Let's make it a reusable component so it will be the same across blocks.

Here is the latest design for it:

image

It's just the logo and a separator for now. It's the same SVG used in the Jetpack dashboard. I think we can further work out what we place here and, as I elaborate on here, p6TEKc-23j , we could potentially put more information on it specific to each block.

Let's get it added to these blocks:
- [ ] Markdown
- [ ] Tiled Galleries
- [ ] Related Posts

@mapk and @thomasguillot I would love some feedback and ideas from you to around this and what we could do in future iterations.

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

* Introduce a new `/components` directory in `/gutenberg` that we can use for reusable components that make sense only in Gutenberg context.
* Introduce a new `JetpackBranding` component that renders a `JetpackLogo` with an `<hr />` above, using `InspectorControls`.
* Add an ESLint ignore rule for the new directory we store this component in.
*  Add Jetpack branding to Markdown block
*  Add Jetpack branding to Related Posts block
*  Add Jetpack branding to Tiled Gallery block

#### Screenshots

Markdown:

Related Posts:

Tiled Gallery:

#### Testing instructions

* Spin up a JN site with latest Jetpack, Gutenberg+Gutenpack and this Calypso branch: https://jurassic.ninja/create/?shortlived&gutenpack&gutenberg&calypsobranch=add/gutenberg-jetpack-branding&jetpack-beta (gutenpack-jn)
* Connect the site.
* Start writing a new post.
* Insert any of the following blocks and select it, then verify branding appears as shown on the screenshots:
  * Markdown
  * Related Posts
  * Tiled Gallery

Fixes #27773

### CODE REVIEW

File: client/gutenberg/components/jetpack-branding/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Internal dependencies
+ */
+import JetpackLogo from 'components/jetpack-logo';
+import './style.scss';
+
+const JetpackBranding = () => (
+	<InspectorControls>
+		<div className="jetpack-branding">
```
Comment (sirreal) [inline client/gutenberg/components/jetpack-branding/index.js]: Can we set the `className` directly on `InspectorControls`?

File: client/gutenberg/components/jetpack-branding/index.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/** @format */
```
Comment (sirreal) [inline client/gutenberg/components/jetpack-branding/index.js]: This is specific to the editor, should the file/Component name reflect that?

**Comment by drw158** [comment]
I believe a lot of design discussion around this has already happened, but the branding just seems _way_ too obtrusive. It feels like an ad to me. The color, the size... it all seems like too much.

File: client/gutenberg/components/jetpack-branding/index.js (hunk 1 above)
Comment (tyxla) [reply client/gutenberg/components/jetpack-branding/index.js]: No, we can't - which is why I did it that way.

**Comment by MichaelArestad** [comment]
I think we have to site on this for the time being sadly. Thanks for putting this together.

**Comment by tyxla** [comment]
No worries - it was a good experiment, as we saw a use case for reusable components that make sense only in gutenberg context in Calypso.

Should we close it for the time being @MichaelArestad?

**Comment by MichaelArestad** [comment]
Yes. Just read an update around that. Let's close this for now. Thanks Marin!

**Pull request closed without merging**